File: app/src/main/java/com/example/modulo5_recycleviewimg/Adapter.java
package com.example.modulo5_recycleviewimg;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;

import java.util.List;

// paso 3 extender de recyclerView
public class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {

    // 1 representacion a los datos
    private List<datosLista> dataList;

    // paso 6 crar interface escuche cuando hace un click


    public Adapter(List<datosLista>dataList){
        this.dataList=dataList;
    }

    // paso 4 inflar la vista del recyclerview
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list, parent, false);
        return new ViewHolder(view);
    }
    //paso 5
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        datosLista data = dataList.get(position);

        holder.textView.setText(data.getDato());
        Glide.with(holder.itemView).load(data.getUrl()).into(holder.imageView);
        // paso 7 agregar escuchador
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
         int position = holder.getBindingAdapterPosition();
                if (position != RecyclerView.NO_POSITION) {
                    datosLista eleccion = dataList.get(position);
                    Bundle bundle = new Bundle();
                    bundle.putString("url", eleccion.getUrl());
                    bundle.putString("datos", eleccion.getDato());

                    SecondFragment fragment = new SecondFragment();
                    fragment.setArguments(bundle);
                    FragmentManager fragmentManager = ((FragmentActivity) view.getContext()).getSupportFragmentManager();
                    fragmentManager.beginTransaction()
                            .replace(R.id.fragmentContainerView, fragment)
                            .addToBackStack(null)
                            .commit();
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }


// 2 crear clase interna Viewholder

    public static class ViewHolder extends RecyclerView.ViewHolder {

        TextView textView;
        ImageView imageView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            textView = itemView.findViewById(R.id.textView2);
            imageView = itemView.findViewById(R.id.imageView2);

        }
    }


}

File: app/src/main/java/com/example/shoestap/Adapter.kt
package com.example.shoestap

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.android.material.floatingactionbutton.FloatingActionButton

class Adapter(private val dataList: List<datosLista>) : RecyclerView.Adapter<Adapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_list, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val data = dataList[position]

        holder.textView.text = data.dato
        Glide.with(holder.itemView).load(data.url).into(holder.imageView)

        holder.itemView.setOnClickListener {
            val position = holder.bindingAdapterPosition
            if (position != RecyclerView.NO_POSITION) {
                val eleccion = dataList[position]
                val bundle = Bundle().apply {
                    putString("url", eleccion.url)
                    putString("datos", eleccion.dato)
                    putString("precio", eleccion.precio)
                    putString("pDescription", eleccion.pDescription)
                }

                val fragment = SecondFragment()
                fragment.arguments = bundle
                val fragmentManager = (holder.itemView.context as FragmentActivity).supportFragmentManager
                fragmentManager.beginTransaction()
                    .replace(R.id.fragmentContainerView, fragment)
                    .addToBackStack(null)
                    .commit()
            }
        }

        holder.fab.setOnClickListener {
            val position = holder.bindingAdapterPosition
            if (position != RecyclerView.NO_POSITION) {
                val eleccion = dataList[position]
                val bundle = Bundle().apply {
                    putString("url", eleccion.url)
                    putString("datos", eleccion.dato)
                    putString("precio", eleccion.precio)
                    putString("pDescription", eleccion.pDescription)
                }

                val fragment = ThirdFragment()
                fragment.arguments = bundle
                val fragmentManager = (holder.fab.context as FragmentActivity).supportFragmentManager
                fragmentManager.beginTransaction()
                    .replace(R.id.fragmentContainerView, fragment)
                    .addToBackStack(null)
                    .commit()
            }
        }
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textView: TextView = itemView.findViewById(R.id.tv_nombre_articulo)
        val imageView: ImageView = itemView.findViewById(R.id.imageView2)
        val fab: FloatingActionButton = itemView.findViewById(R.id.fab)
    }
}
